### PR TITLE
Minor change to http:options and path-combine

### PR DESCRIPTION
**Pull request by owaddell**
This pull request
- adds `validate-path` to `http:options` and exposes `http:valid-path?` so a server can supply a `http:url-handler` that interprets paths that would otherwise be invalid, yet still check the extracted paths before hitting the filesystem, and
- modifies `path-combine` to avoid adding extra directory separators while allocating less in some cases.
- [edit] fixes a bug in `http:eval-dynamic-page`

### CODE REVIEW

File: doc/swish/http.tex [hunk 1]
```diff
@@ -303,7 +303,12 @@ \subsection{WebSocket Protocol}\label{sec:websocket-protocol}
 
   \code{file-transform} & \code{(lambda (path) \#f)} & a procedure; a
   cache process calls this procedure with a file path and expects an
-  evaluator procedure or \code{\#f} (see \S\ref{sec:dynamic-pages})
+  evaluator procedure or \code{\#f} (see \S\ref{sec:dynamic-pages}) \\
+
+  \code{validate-path} & \code{http:valid-path?} & a predicate;
+  the server raises an exception if the \code{<request>} \var{path}
+  does not satisfy this prediate. \\
```
Comment (burgerrg) [inline doc/swish/http.tex]: prediate => predicate

File: doc/swish/http.tex [hunk 2]
```diff
@@ -682,6 +687,16 @@ \subsection{WebSocket Protocol}\label{sec:websocket-protocol}
 Section~\ref{sec:dynamic-page-constructs}, and calls the \code{eval}
 system procedure.
 
+\defineentry{http:valid-path?}
+\begin{procedure}
+  \code{(http:vavlid-path? \var{path})}
```
Comment (burgerrg) [inline doc/swish/http.tex]: vavlid => valid

File: doc/swish/http.tex (hunk 1 above)
Comment (owaddell) [reply doc/swish/http.tex]: Oh dear, thanks for finding this. Fixed.

File: doc/swish/http.tex (hunk 2 above)
Comment (owaddell) [reply doc/swish/http.tex]: Thanks for finding this. Fixed.

**Review by burgerrg — APPROVED** [review]

**Review by laqrix — APPROVED** [review]
Nice work Oscar!

**Comment by owaddell** [comment]
Thanks, guys.

I noticed a failed macOS test and pushed an update to use `test:receive` in that mat.

**Comment by owaddell** [comment]
Oops. One more wafer-thin commit.

**Comment by owaddell** [comment]
Thanks for reviewing this, gentlemen.

**Pull request merged**